### PR TITLE
Move SCTP to GA

### PR DESCRIPTION
**Pull request by danwinship**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Moves SCTP to GA, locks the feature gate on, removes feature-gate checks from relevant code

**Special notes for your reviewer**:
I wasn't sure if I should leave behind the empty conditional validation functions (so it's easier to start using them again later) or if I should remove them (which would then require updating the corresponding strategy files to not call them)...

**Does this PR introduce a user-facing change?**:
```release-note
SCTP support in API objects (Pod, Service, NetworkPolicy) is now GA.
Note that this has no effect on whether SCTP is enabled on nodes at the kernel level,
and note that some cloud platforms and network plugins do not support SCTP traffic.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
- [Enhancement]: https://github.com/kubernetes/enhancements/issues/614
- [KEP]: https://github.com/kubernetes/enhancements/tree/ee5e90d5/keps/sig-network/614-SCTP-support
```

/sig network
/priority important-soon

### CODE REVIEW

File: pkg/apis/core/validation/conditional_validation.go [hunk 1]
```diff
@@ -115,44 +109,9 @@ func joinIPFamilies(families []api.IPFamily, separator string) string {
...
-	for pIndex, p := range service.Spec.Ports {
-		if p.Protocol == api.ProtocolSCTP {
-			fields = append(fields, field.NewPath("spec.ports").Index(pIndex).Child("protocol"))
-		}
-	}
-	return fields
-}
-
 // ValidateConditionalEndpoints validates conditionally valid fields.
 func ValidateConditionalEndpoints(endpoints, oldEndpoints *api.Endpoints) field.ErrorList {
```
Comment (thockin) [inline pkg/apis/core/validation/conditional_validation.go:113]: why not nix the whole function?

File: pkg/apis/core/validation/conditional_validation.go [hunk 2]
```diff
@@ -188,41 +147,5 @@ func ValidateConditionalPod(pod, oldPod *api.Pod, fldPath *field.Path) field.Err
 }
 
 func validateConditionalPodSpec(podSpec, oldPodSpec *api.PodSpec, fldPath *field.Path) field.ErrorList {
```
Comment (thockin) [inline pkg/apis/core/validation/conditional_validation.go:149]: same?

File: pkg/apis/networking/validation/conditional_validation.go [hunk 3]
```diff
@@ -18,42 +18,10 @@ package validation
...
 import (
 	"k8s.io/apimachinery/pkg/util/validation/field"
-	utilfeature "k8s.io/apiserver/pkg/util/feature"
-	api "k8s.io/kubernetes/pkg/apis/core"
 	"k8s.io/kubernetes/pkg/apis/networking"
-	"k8s.io/kubernetes/pkg/features"
 )
 
 // ValidateConditionalNetworkPolicy validates conditionally valid fields.
 func ValidateConditionalNetworkPolicy(np, oldNP *networking.NetworkPolicy) field.ErrorList {
```
Comment (thockin) [inline pkg/apis/networking/validation/conditional_validation.go:25]: same?

Comment (thockin) [reply pkg/apis/networking/validation/conditional_validation.go:25]: in fact - whole file?

**Comment by danwinship** [comment]
Yeah, I mentioned that in the "notes for your reviewer"; I wasn't sure if maybe it was better to keep them around in case we needed them again later? But I can kill them...

**Comment by liggitt** [comment]
I'm in favor of dropping the conditional validation methods we don't need. That was somewhat undesirable to begin with.

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**